students/hirot_directory/session10/oo_mailroom_hiro/mailroom.py

```diff
@@ -0,0 +1,212 @@
+                       "(4) quit\n"))
+    return answer
+
+
+def select_quit():
+    selection = (input("Are you sure to quit? (Y or N) To return to main menu, please select 'N'.: ")).upper()
+    return selection
+
+
+def mainloop():
+    while True:
+        try:
+            answer = select_mainmenu()
+            dict_answer[answer]()
+
+        except ValueError as e:
+            print("Error happened. Error is ", type(e), "Please type 1, 2 or 3")
+        except KeyError as k:
+            print("Error happened. Error is ", type(k), "Please type 1, 2 or 3")
+
+
+def thank_you_loop():
+    while True:
+        try:
+            choice = select_submenu()
+            dict_choice[choice]()
+
+        except ValueError as e:
+            print("Error happened. Error is ", type(e) , "Please type 1, 2 or 3")
+        except KeyError as k:
+            print("Error happened. Error is ", type(k) , "Please type 1, 2 or 3")
+
+
+def quit():
+    print("quit function is called")
+
+    try:
+        selection = select_quit()
+        if (selection == 'Y'):
+            print("Good bye...")
+            sys.exit()
+        else:
+            return mainloop()
+
+    except ValueError as e:
+        print("Error happened. Error is ", type(e), "Please type 1, 2 or 3")
+    except KeyError as k:
+        print("Error happened. Error is ", type(k), "Please type 1, 2 or 3")
+
+
+def make_sample_db():
+
+    donor_names = ["William Gates III", "Mark Zuckerberg", "Jeff Bezos", "Paul Allen"]
+    donations = [[200, 500], [2000, 3000], [4000, 5000], [100, 150]]
+
+    sample_db = {}
+
+    for name, donation in zip(donor_names, donations):
+        sample_db[name] = donation
+
+    return sample_db
+
+    # return [Donor("William Gates III", [653772.32, 12.17]),
+    #         Donor("Jeff Bezos", [877.33]),
+    #         Donor("Paul Allen", [663.23, 43.87, 1.32]),
+    #         Donor("Mark Zuckerberg", [1663.23, 4300.87, 10432.0]),
+    #         ]
+
+
+def find_donor(name):
+    if name in make_sample_db().sample_db.keys():
+        return name
+    else:
+        return None
+
+    # return self.donor_data.get(Donor.normalize_name(name))
+
+# db = DonorDB(make_sample_db())
+
+
+class DonorDB():
+
+    def __init__(self):
+
+        self.all_donors = {}
+
+        # self.all_donors = [Donor(name, donations)
+        #                    for name, donations in zip(self.donor_names, self.donations)]
+
+
+
+
+    def add_donor(self, name):
+        donor = Donor(name)
+        self.all_donors[name] = donor
+        return donor
+
+
+    def print_list(self):
+
+        print("{msg1: <50}| {msg2: <12}| "
+
+              "{msg3:<10}| {msg4: <12}".format(msg1="Donor Name",
+
+                                               msg2="Total Given",
+
+                                               msg3="Num Gifts",
+
+                                               msg4="Average Gift"))
+        for donor in self.all_donors:
+            for key in donor.donor_dict:
```

Comment:
the Donor object doesn't (or shouldn't) have a donor_dict -- it has a name and donations....